server/channels/app/status.go

```diff
@@ -77,6 +77,13 @@ func (a *App) SetCustomStatus(c request.CTX, userID string, cs *model.CustomStat
 		return model.NewAppError("SetCustomStatus", "api.custom_status.set_custom_statuses.update.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	// Ensure the emoji exists before saving the custom status even if it's deleted afterwards
+	if cs.Emoji != "" {
+		if err := a.confirmEmojiExists(c, cs.Emoji); err != nil {
```

Comment:
@hmhealey have we tested it with a very large emoji name to make sure that it doesn't cause an issue while trying to validate if it exists? Ticket had the example of an emoji name with e.g. 100000 characters

Reply:
Yeah, I tested that case. It returns pretty quickly because we check the length of the name before doing anything more complex like checking the DB to see if the emoji exists